### PR TITLE
[C#] fix: Remove `[]` intializer notation

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #minor

## Details
remove `[]` notation because it is not supported in version < C#12. That includes .NET 6 which is a framework we target.

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by corinagum** [comment]
Is there a reason why there are so many commits within the PR?

**Comment by singhk97** [comment]
> Is there a reason why there are so many commits within the PR?

Because I'm reusing the `dotnet-dev` branch and it hadn't been merged into main yet.

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/StreamingChatToolCallsBuilder.cs [hunk 1]
```diff
@@ -5,9 +5,9 @@ namespace Microsoft.Teams.AI.AI.Models
 {
     public class StreamingChatToolCallsBuilder
```
Comment (corinagum) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/StreamingChatToolCallsBuilder.cs:6]: Some warnings here, could you resolve?

Comment (corinagum) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/StreamingChatToolCallsBuilder.cs:6]: Looks like this file has a bunch

Comment (singhk97) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/StreamingChatToolCallsBuilder.cs]: resolved

**Review by corinagum — APPROVED** [review]
LGTM~

**Pull request merged**